server/data/restClientResponseMiddleware.ts

```diff
@@ -0,0 +1,18 @@
+/* istanbul ignore file */
+
+import { Response, SuperAgentRequest } from 'superagent'
+import inMemoryStore from '../inMemoryStore'
+
+function restClientResponseMiddleware(agent: SuperAgentRequest) {
+  agent.on('request', ({ req }) => {
+    req.on('response', (res: Response, err: Error) => {
```

Comment:
Might `userVersionMiddleware` be a more accurate name? 
A lot of middleware could deal with responses, so it could be helpful to be more specific